File: Java/shatteredcake.java
/*
 *	Problem: "Shattered Cake"
 */

import java.util.Scanner;

public class shatteredcake {
	
	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);
		int width = sc.nextInt();
		int numPieces = sc.nextInt();
		
		int totalArea = 0;
		for (int i = 0; i < numPieces; i++) {
			int widthPiece = sc.nextInt();
			int lengthPiece = sc.nextInt();
			totalArea += widthPiece * lengthPiece;
		}
		
		int length = totalArea / width;
		System.out.println(length);		
	}
	
}

File: Python/shatteredcake.py
'''
 *
 *	Problem: "Shattered Cake"
 *
'''

import sys
import math

width = int(input())
numPieces = int(input())

totalArea = 0
for x in range(numPieces):
	
	widthPiece, lengthPiece = (int(dim) for dim in input().split())
	totalArea += widthPiece * lengthPiece
	
length = int(totalArea / width)
print(length)